pkg/misc/slice/chunk.go

```diff
@@ -0,0 +1,16 @@
+package slice
+
+func Chunk[T any](slice []T, maxChunkSize int, cb func([]T) error) error {
+	for start := 0; start < len(slice); start += maxChunkSize {
+		end := start + maxChunkSize
+		if end > len(slice) {
+			end = len(slice)
+		}
+
+		if err := cb(slice[start:end]); err != nil {
+			return err
```

Comment:
nit: Do we need a more descriptive error here?

Reply:
Fixed in 768efffea0160274ec0f6672d3c2ba5692ab9698